In [1]:
import pandas as pd

In [2]:
sales_data=pd.read_csv("../../Datasets/sales.csv/sales.csv")
sales_data.describe()
sales_data.head(1000)

date  id  category_id  sales  views  price_cost  price_retail
0    2022-02-24   1            3      0      0           0             0
1    2022-02-25   1            3      0      0           0             0
2    2022-02-26   1            3      0      0           0             0
3    2022-02-27   1            3      0      0           0             0
4    2022-02-28   1            3      0      0           0             0
..          ...  ..          ...    ...    ...         ...           ...
995  2022-11-14   2            2      0    143          60            78
996  2022-11-15   2            2      0    249          60            78
997  2022-11-16   2            2      0    337          60            78
998  2022-11-17   2            2      0    446          60            78
999  2022-11-18   2            2     50    394          60            78

[1000 rows x 7 columns]

<div>
The Data provided is related to Bike prices on a particular date
    <ul>
        <li>id: Id of the bike</li>
        <li>category_id: Category of the bike</li>
        <li>sales: How many sold on the particular day</li>
        <li>views: How many bikes are viewed that day</li>
        <li>price_cost: What is the cost of price</li>
        <li>price_retail: What is the retail price for selling</li>
    </ul>
</div>

In [4]:
print(sales_data.id.unique().size)
print(sales_data.category_id.unique().size)

3482
4


<ol>
    <li>There are a total of 3482 bikes available</li>
    <li>There are 4 categories in which all these 3482 bikes comes under</li>
</ol>

<b><h4>Question</h4></b>
<b>1.Each category has how many bikes, is there any bikes that are part of more than one category</b>


In [5]:
sales_data_category_id_grouped=sales_data.groupby('category_id')
print(sales_data_category_id_grouped.size())

category_id
1    1263432
2    1076040
3     198372
4      10980
dtype: int64


In [6]:
id_cateogory_separateData=[]
for i in range(1,5):
    id_cateogory_separateData.append(sales_data_category_id_grouped.get_group(i))

id_category_1=pd.Series(sales_data.loc[sales_data.category_id==1].id.unique())
id_category_2=pd.Series(sales_data.loc[sales_data.category_id==2].id.unique())
id_category_3=pd.Series(sales_data.loc[sales_data.category_id==3].id.unique())
id_category_4=pd.Series(sales_data.loc[sales_data.category_id==4].id.unique())

In [7]:
for i in id_cateogory_separateData:
    print(i.id.unique().size)

for i in range(len(id_cateogory_separateData)):
    for j in range(i+1,len(id_cateogory_separateData)):
        print("%d %d: %s"%(i+1,j+1,True in (id_cateogory_separateData[i].isin(id_cateogory_separateData[j]))),end=":")
        print('')

1726
1470
271
15
1 2: False:
1 3: False:
1 4: False:
2 3: False:
2 4: False:
3 4: False:


<i><h4>Answer:</h4></i>
<i>1. Category_1: 1726, Category_2:1470, Category_3=271, Category_4: 15. And there are no bikes that belongs to more than one Category</i>

Category 3 and 4 has very less bikes compared to 1 and 2

<b><h4>Question</h4></b>
<b>Identify Which group has the highest sales and which group has the lowest sales</b>

In [8]:
for i in range(1,5):
    print("%d sales count:%d"%(i,id_cateogory_separateData[i-1].sales.sum()))

for i in range(1,5):
    print("%d views count:%d"%(i,id_cateogory_separateData[i-1].views.sum()))


1 sales count:127308
2 sales count:743416
3 sales count:73233
4 sales count:656
1 views count:53906343
2 views count:34839778
3 views count:8343857
4 views count:405849


Category 2 has the highest sales and category 4 has the least sales
Category 1 has the highest views and category 4 has the lowest views

In [9]:
print("Min Date: %s"%(sales_data.date.min()))
print("Max Date: %s"%(sales_data.date.max()))
print("Date Data Description:")
print(sales_data.date.describe())

Min Date: 2022-02-24
Max Date: 2024-02-25
Date Data Description:
count        2548824
unique           732
top       2022-02-24
freq            3482
Name: date, dtype: object


# Filtered Data Section

In [10]:
filtered_data=sales_data[sales_data['price_cost']!=0]
filtered_data.head()
filtered_data.id.value_counts()

id
796     732
898     732
905     732
904     732
903     732
       ... 
1520      7
1316      6
2798      6
288       5
3000      5
Name: count, Length: 3463, dtype: int64

In [11]:
filtered_data_by_category=filtered_data.groupby('category_id')
print(filtered_data_by_category['price_cost'].max())

category_id
1    158951
2     39900
3      4489
4    292573
Name: price_cost, dtype: int64


In [12]:
filtered_data.sort_values(by='price_cost',ascending=False).head()
filtered_unique_data=filtered_data.drop_duplicates()